storage/flux/table.gen.go.tmpl

```diff
@@ -706,26 +707,93 @@ func (t *{{.name}}GroupTable) Do(f func(flux.ColReader) error) error {
 }
 
 func (t *{{.name}}GroupTable) advance() bool {
-RETRY:
-	a := t.cur.Next()
-	l := a.Len()
-	if l == 0 {
-		if t.advanceCursor() {
-			goto RETRY
-		}
-
+	if t.cur == nil {
+		// For group aggregates, we will try to get all the series and all table buffers within those series
+		// all at once and merge them into one row when this advance() function is first called.
+		// At the end of this process, t.advanceCursor() already returns false and t.cur becomes nil.
+		// But we still need to return true to indicate that there is data to be returned.
+		// The second time when we call this advance(), t.cur is already nil, so we directly return false.
 		return false
 	}
+	var arr *cursors.{{.Name}}Array
+	var len int
+	for {
+		arr = t.cur.Next()
+		len = arr.Len()
+		if len > 0 {
+			break
+		}
+		if !t.advanceCursor() {
+			return false
+		}
+	}
 
-	// Retrieve the buffer for the data to avoid allocating
-	// additional slices. If the buffer is still being used
-	// because the references were retained, then we will
-	// allocate a new buffer.
-	cr := t.allocateBuffer(l)
-	cr.cols[timeColIdx] = arrow.NewInt(a.Timestamps, t.alloc)
-	cr.cols[valueColIdx] = t.toArrowBuffer(a.Values)
-	t.appendTags(cr)
-	t.appendBounds(cr)
+	// handle the group without aggregate case
+	if t.gc.Aggregate() == nil {
+		// Retrieve the buffer for the data to avoid allocating
+		// additional slices. If the buffer is still being used
+		// because the references were retained, then we will
+		// allocate a new buffer.
+		colReader := t.allocateBuffer(len)
+		colReader.cols[timeColIdx] = arrow.NewInt(arr.Timestamps, t.alloc)
+		colReader.cols[valueColIdx] = t.toArrowBuffer(arr.Values)
+		t.appendTags(colReader)
+		t.appendBounds(colReader)
+		return true
+	}
+
+	// handle the group with aggregate case
+	var value {{.Type}}
+	// For group count, sum, min, and max, the timestamp here is always math.MaxInt64.
+	// their final result does not contain _time, so this timestamp value can be anything
+	// and it won't matter.
+	// For group first, we need to assign the initial value to math.MaxInt64 so
+	// we can find the row with the smallest timestamp.
+	// Do not worry about data with math.MaxInt64 as its real timestamp.
+	// In OSS we require a |> range() call in the query and a math.MaxInt64 timestamp
+	// cannot make it through.
+	var timestamp int64 = math.MaxInt64
+	if t.gc.Aggregate().Type == datatypes.AggregateTypeLast {
+		timestamp = math.MinInt64
+	}
+	for {
+		// note that for the group aggregate case, len here should always be 1
+		for i := 0; i < len; i++ {
+			switch t.gc.Aggregate().Type {
+			case datatypes.AggregateTypeCount:
+				{{if eq .Name "Integer"}}fallthrough{{else}}panic("unsupported for aggregate count: {{.Name}}"){{end}}
```

Comment:
I don't think I follow the logic here. Every datatype should support count right?

Reply:
This is for re-aggregating `count`. The counts at this stage can only be integers.

Reply:
Got it. Thanks for clarifying.